types/src/transaction/mod.rs

```diff
@@ -366,6 +368,10 @@ pub enum RawTransactionWithData {
         raw_txn: RawTransaction,
         secondary_signer_addresses: Vec<AccountAddress>,
     },
+    MultiAgentWithFeePayer {
+        raw_txn: RawTransaction,
+        secondary_signer_addresses: Vec<AccountAddress>,
```

Comment:
should this not include an additional signer called `fee_payer: Option<AccountAddress>` or maybe no option?

Reply:
It's still not clear how to get fee_payer into the system. But we can always treat it as just another agent

Reply:
You mean `TransactionAuthenticator`? Yeah, I think we need a new one of these, this is still hacky.